pkg/space/space.go

```diff
@@ -59,5 +61,15 @@ func NewSubSpace(spaceRequest *toolchainv1alpha1.SpaceRequest, parentSpace *tool
 
 // SubSpaceName generates a name for a subSpace based on parentSpace name and spacerequest UID.
 func SubSpaceName(parentSpace *toolchainv1alpha1.Space, spacerequest *toolchainv1alpha1.SpaceRequest) string {
-	return parentSpace.GetName() + string(spacerequest.UID[:5]) // we only get the first 5 chars from UID so we can keep the name within the length limits
+	parentSpaceName := parentSpace.GetName()
+
+	// if the parent space is itself a subspace, then we need to strip its
+	// identifier out to prevent length limitations from kicking in
+	if parentSpace.Spec.ParentSpace != "" && len(parentSpaceName) > 6 {
+		// take of 6 characters to include the dash
```

Comment:
minor thing - maybe adding a small example to make it clear to the reader that we are removing the UID suffix from the parent space name before using it for the sub space name creation. WDYT ? 
```suggestion
		// take of 6 characters to include the dash
		// for example if the parentSpaceName is "parentSpace-01234" then we keep only "parenSpace", removing the 5char UID suffix from the end. 
```

Reply:
sounds good to me - I've adjusted some of the comments with a7f983f, hopefully things are a bit clearer.

Reply:
looks good ! thanks !